npm/util/ioutil/grep.go

```diff
@@ -2,10 +2,69 @@
 
 package ioutil
 
-import utilexec "k8s.io/utils/exec"
+import (
+	"fmt"
+
+	utilexec "k8s.io/utils/exec"
+)
 
 // Grep is the grep command string
-const Grep = "grep"
+const (
+	Grep                 = "grep"
+	GrepRegexFlag        = "-P"
+	GrepOnlyMatchingFlag = "-o"
+	GrepBeforeFlag       = "-B"
+)
+
+func DoublePipeToGrep(command, grepCommand1, grepCommand2 utilexec.Cmd) (searchResults []byte, gotMatches bool, commandError error) {
+	pipe, commandError := command.StdoutPipe()
+	if commandError != nil {
+		return
+	}
+	grepCommand1.SetStdin(pipe)
+
+	grepPipe, commandError := grepCommand1.StdoutPipe()
+	if commandError != nil {
+		_ = pipe.Close()
+		commandError = fmt.Errorf("error getting pipe for grep command: %w", commandError)
+		return
+	}
+	grepCommand2.SetStdin(grepPipe)
+
+	closePipes := func() {
+		_ = grepPipe.Close()
+		_ = pipe.Close()
+	}
+	defer closePipes()
+
+	commandError = grepCommand1.Start()
+	if commandError != nil {
+		commandError = fmt.Errorf("error while starting first grep command: %w", commandError)
+		return
+	}
+
+	commandError = command.Start()
+	if commandError != nil {
+		commandError = fmt.Errorf("error while starting command: %w", commandError)
+		_ = grepCommand1.Wait()
+		return
+	}
+
+	waitForAll := func() {
+		_ = command.Wait()
+		_ = grepCommand1.Wait()
+	}
+	defer waitForAll()
+
+	output, err := grepCommand2.CombinedOutput()
+	if err != nil {
+		// grep returns err status 1 if nothing is found
```

Comment:
Can we log this error, just in case if there is something else wrong here ?

Reply:
there won't be any valuable information. Trying this locally in a main.go file, we only get info about the final grep command. All other info (like if the first command failed) is lost